test/ProperLogger.test.ts

```diff
@@ -0,0 +1,494 @@
+/* eslint-disable no-console, @typescript-eslint/no-empty-function */
+
+import { ProperLogger, LogLevels } from '..';
+
+describe('LogLevels', () => {
+  test('logging levels correspond to acending values as severity increases', () => {
+    expect(LogLevels.DEBUG).toEqual(10);
+    expect(LogLevels.INFO).toEqual(20);
+    expect(LogLevels.WARNING).toEqual(30);
+    expect(LogLevels.ERROR).toEqual(40);
+  });
+});
+
+describe('ProperLogger', () => {
+  function setupTestLogger(): ProperLogger {
+    const logger = new ProperLogger('test');
+    logger.console = {
+      debug: jest.fn(),
+      log: jest.fn(),
+      warn: jest.fn(),
+      error: jest.fn(),
+    };
+
+    return logger;
+  }
+
+  describe('debug', () => {
+    test('only message', () => {
+      const logger = setupTestLogger();
+      logger.logLevel = LogLevels.DEBUG;
+
+      expect(logger.console.debug).not.toHaveBeenCalled();
+
+      logger.debug('I think there be six Richmonds in the field;');
+
+      expect((logger.console.debug as jest.Mock).mock.calls).toEqual([
```

Comment:
Is there a reason you didn't use [`toHaveBeenCalledWith`](https://jestjs.io/docs/en/expect#tohavebeencalledwitharg1-arg2-) here? Issue with TypeScript?

Reply:
Not sure I follow, but maybe the answer is [`toHaveBeenNthCalledWith`](https://jestjs.io/docs/en/expect#tohavebeennthcalledwithnthcall-arg1-arg2-)?

Reply:
Ah yep that works. :thumbsup: